### PR TITLE
Important modification to network deserialization

### PR DESCRIPTION
**Pull request by milesial**
Hi, this small PR has some important objectives.

1. First, the behaviour of `torch.load`, `torch.save` uses `pickle`, so the whole instance is saved, no need to save / load the state dict and optimizer separately.

2. More importantly, this allows student to submit **custom policies** to the evaluator (policies that are not part of this repo, with different architectures).
Without that, students are required to open a Pull Request here each time they wish to use a policy that is not part of the original code (see https://github.com/osigaud/Basic-Policy-Gradient-Labs/pull/3 ).

As a reminder, the lab instructions state:
> The nature of the policies themselves is open to students
>  Besides, if they use a network, its precise architecture (number oflayers, number of neurons, transfer functions, etc.) is open.

### For students: how to use custom policies

The custom policy has to be saved as TorchScript, and the `torch.load` will call `torch.jit.load`, which will load the compiled model as a `RecursiveScriptModule` instead of a `nn.Module`.

The loaded module can have the `select_action` method, so the evaluator code works as usual.

Example of using and saving a custom (deterministic) policy (on the students' side):

```python
from typing import List

class ExamplePolicy(nn.Module):
    def __init__(self, l1, l2, l3, l4):
        super().__init__()
        self.pi = nn.Sequential(
            nn.Linear(l1, l2),
            nn.ReLU(),
            nn.Linear(l2, l3),
            nn.ReLU(),
            nn.Linear(l3, l4),
            nn.Tanh()
        )

    def forward(self, state):
        if not isinstance(state, torch.Tensor):
            state = torch.from_numpy(state).float()
        return self.pi(state)

    @torch.jit.export
    def select_action(self, state: List[float], deterministic: bool=False) -> List[float]:
        state = torch.tensor(state)
        action = self.forward(state)
        act: List[float] = action.data.tolist()
        return act

p = ExamplePolicy(3, 128, 64, 1)
traced = torch.jit.script(p)
torch.jit.save(traced, 'pol/Pendulum-v0#MILESI - MILESI#custom#500#500.zip')
```

On the evaluator side, nothing needs to be changed and the policy will be evaluated as usual.

More info on TorchScript / the JIT:
https://pytorch.org/docs/stable/jit_language_reference.html#language-reference
https://pytorch.org/docs/stable/jit.html?highlight=jit#creating-torchscript-code

### CODE REVIEW

**Comment by osigaud** [comment]
Thanks a lot for proposing this very useful improvement. I have hesitated before taking it because my students are currently working on the code with a deadline by the end of november, but I asked them and this will definitely simplify their work.
I just hope that pytorch 1.6.0 is also available for windows and MacOS.
Best regards,
Olivier Sigaud
